modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IO.kt

```diff
@@ -593,11 +604,11 @@ sealed class IO<out E, out A> : IOOf<E, A> {
    *
    * **BEWARE** this does **not** support cancelation since Kotlin has no cancelation support for `suspend` on the language level.
    */
-  suspend fun suspended(): A = suspendCoroutine { cont ->
+  suspend fun suspended(): Either<E, A> = suspendCoroutine { cont ->
```

Comment:
This might be non sense but, shouldn't we have a suspended version for the `IO` alias that still returns `A` (or throws)?

Reply:
Well that is a very good question! :D We have a problem with distinct-ing between `IO<Nothing, A>` and `IO<E, A>`since due to erasure they're the same. Which leaves us with the following, same problem with the other runners.

```
fun <E, A> IOOf<E, A>.suspended(): Either<E, A>
fun Either<Nothing, A>.value(): A = fold(::identity, ::identity)
fun <A> IOOf<Nothing, A>.suspendedGet(): A = fix().suspended().value()
```

Reply:
So what's on the left side if you just do `suspended()` for an `IO<A>`?

Reply:
`Nothing` so you can safely extract `A` using `fold(::identity, ::identity)`

Reply:
I mean, `Nothing` is the type but what's the value of the left side in that case? Mostly thinking on what's gonna happen if they try to read it.

Reply:
If the type is `Nothing` then there can be no value on the left side. Or in other words `Either<Nothing, A>` is isomorphic to `Id<A>` and `A`.